lib/active_merchant/billing/gateways/securion_pay.rb

```diff
@@ -0,0 +1,269 @@
+        commit(:post, 'charges', post, options)
+      end
+
+      def capture(money, authorization, options = {})
+        post = {}
+        add_amount(post, money, options)
+
+        commit(:post, "charges/#{CGI.escape(authorization)}/capture", post, options)
+      end
+
+      def refund(money, authorization, options = {})
+        post = {}
+        add_amount(post, money, options)
+        commit(:post, "charges/#{CGI.escape(authorization)}/refund", post, options)
+      end
+
+      def void(authorization, options = {})
+        commit(:post, "charges/#{CGI.escape(authorization)}/refund", {}, options)
+      end
+
+      def verify(credit_card, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def store(payment, options = {})
+        post = {}
+        add_creditcard(post, payment, options)
+
+        post[:description] = options[:description] if options[:description]
+        post[:email] = options[:email] if options[:email]
+        if options[:customer]
+          update_customer(options[:customer], post, options)
+        else
+          MultiResponse.run(:first) do |r|
+            r.process { verify(payment, options) }
+            if r.params["id"].present?
+              r.process { commit(:post, "customers", post, options) }
+            end
+          end
+        end
+      end
+
+      def update_customer(customer_id, post, options = {})
+        commit(:post, "customers/#{CGI.escape(customer_id)}", post, options)
+      end
+
+      def update(customer_id, card_id, options = {})
+        commit(:post, "customers/#{CGI.escape(customer_id)}/cards/#{CGI.escape(card_id)}", options, options)
+      end
+
+      def unstore(customer_id, options = {})
+        if options[:card_id]
+          commit(:delete, "customers/#{CGI.escape(customer_id)}/cards/#{CGI.escape(options[:card_id])}", nil, options)
+        else
+          commit(:delete, "customers/#{CGI.escape(customer_id)}", nil, options)
+        end
+      end
+
+      def create_token(payment, options = {})
+        post = {}
+        add_creditcard(post, payment, options)
+
+        commit(:post, "tokens", post[:card], options)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: Basic )\w+), '\1[FILTERED]').
+          gsub(%r((card\[number\]=)\d+), '\1[FILTERED]').
+          gsub(%r((card\[cvc\]=)\d+), '\1[FILTERED]')
+      end
+
+      private
+
+      def add_customer(post, payment, options)
+        post[:customer] = options[:customer] if options[:customer] && !payment.respond_to?(:number)
+      end
+
+      def add_customer_data(post, options)
+        metadata_options = [:description, :ip, :user_agent, :referrer]
+        post.update(options.slice(*metadata_options))
```

Comment:
Lets just expand this out; we prefer explicit to implicit. So assign each thing vs. using `update`/`slice`.